### PR TITLE
[FX-2323] Improves handling of fair hero image

### PR DESCRIPTION
**Pull request by dzucconi**
Closes [FX-2323](https://artsyproduct.atlassian.net/browse/FX-2323)

### CODE REVIEW

File: src/v2/Apps/Fair/Components/FairHeader/FairHeaderImage.tsx [hunk 1]
```diff
@@ -19,18 +19,32 @@ const AspectRatioBox = styled(Box)<{ ratio?: ResponsiveValue<number> }>`
   ${ratioPadding}
 `
 
+const Container = styled(Box)<FontSizeProps>`
+  overflow: hidden;
+  ${fontSize}
```
Comment (dzucconi) [inline src/v2/Apps/Fair/Components/FairHeader/FairHeaderImage.tsx:24]: Oh woops I can remove this now; I was messing around with another way of doing this using font-size and em units but just using viewport units is easier to understand.

File: src/v2/Apps/Fair/Components/FairHeader/FairHeaderImage.tsx [hunk 2]
```diff
@@ -19,18 +19,31 @@ const AspectRatioBox = styled(Box)<{ ratio?: ResponsiveValue<number> }>`
...
+  }
 `
 
 interface FairHeaderImageProps {
   fair: FairHeaderImage_fair
 }
 
+const ASPECT_WIDTH = 3
+const ASPECT_HEIGHT = 4
+const HEIGHT_VH = 70
```
Comment (dzucconi) [inline src/v2/Apps/Fair/Components/FairHeader/FairHeaderImage.tsx:44]: A bit less than 80vh to accommodate the navbar

File: src/v2/Apps/Fair/Components/FairHeader/FairHeaderImage.tsx [hunk 3]
```diff
@@ -57,22 +68,26 @@ export const FairHeaderImage: React.FC<FairHeaderImageProps> = ({
...
+          <picture>
+            <source
+              srcSet={`${image.lg_1x.src} 1x, ${image.lg_2x.src} 2x`}
+              media="(min-height: 900px)"
+            />
 
+            <source srcSet={`${image.md_1x.src} 1x, ${image.md_2x.src} 2x`} />
+
+            <Image src={image.md_1x.src} alt={name} lazyLoad />
+          </picture>
```
Comment (dzucconi) [inline src/v2/Apps/Fair/Components/FairHeader/FairHeaderImage.tsx:88]: Picture tag here so as to swap between md/lg versions of the image according to the height of the viewport.

**Comment by mzikherman** [comment]
🔥

File: src/v2/Apps/Fair/Components/FairHeader/FairHeaderImage.tsx [hunk 4]
```diff
@@ -94,6 +110,19 @@ export const FairHeaderImageFragmentContainer = createFragmentContainer(
...
+          }
+          md_2x: cropped(width: 1200, height: 1600, version: "wide") {
+            src: url
+          }
+          lg_1x: cropped(width: 900, height: 1200, version: "wide") {
+            src: url
+          }
+          lg_2x: cropped(width: 1800, height: 2400, version: "wide") {
+            src: url
+          }
```
Comment (dzucconi) [inline src/v2/Apps/Fair/Components/FairHeader/FairHeaderImage.tsx:125]: I think it's high time I exposed a `srcSet` field in metaphysics, this is getting old fast.

File: src/v2/Apps/ViewingRoom/__tests__/ViewingRoomApp.jest.tsx [hunk 5]
```diff
@@ -375,6 +376,7 @@ describe("ViewingRoomApp", () => {
       expect(wrapper.find("ViewingRoomTabBar").length).toBe(0)
       expect(wrapper.find("ViewingRoomContentNotAccessible").length).toBe(0)
       expect(wrapper.html()).not.toContain("some child")
+      jest.runAllTimers()
       expect(mediator.trigger).toHaveBeenCalled()
```
Comment (dzucconi) [inline src/v2/Apps/ViewingRoom/__tests__/ViewingRoomApp.jest.tsx:380]: Unrelated, but I guess this spec started failing. There's a `setTimeout` wrapping the call to this function on the next tick to deal with a race condition.

**Pull request merged**